prime-router/settings/STLTs/WA/wa-phd.yml

```diff
@@ -0,0 +1,127 @@
+              - "Bundle.entry.resource.ofType(Patient).name.family = unknown"
+              # "doesNotMatch(patient_state,Unknown, unknown, UNKNOWN)"
+              - "Bundle.entry.resource.ofType(Patient).where(address.state = 'UNKNOWN')"
+              - "Bundle.entry.resource.ofType(Patient).where(address.state = 'unknown')"
+              - "Bundle.entry.resource.ofType(Patient).where(address.state = 'Unknown')"
+              # "hasAtLeastOneOf(patient_street,patient_zip_code,patient_phone_number,patient_email)"
+              - "Bundle.entry.resource.ofType(Patient).address.line.empty() and Bundle.entry.resource.ofType(Patient).address.postalCode.empty() and Bundle.entry.resource.ofType(Patient).telecom.empty()"
+              # "hasAtLeastOneOf(order_test_date,specimen_collection_date_time,test_result_date)"
+              - "Bundle.entry.resource.ofType(DiagnosticReport).where( basedOn.resolve().extension('https://reportstream.cdc.gov/fhir/StructureDefinition/effective-date').empty() and specimen.resolve().collection.collected.empty() and effective.empty() and issued.empty() and result.resolve().effective.empty()).count() = 0"
+              # "hasValidDataFor(message_id,equipment_model_name,specimen_type,test_result,patient_last_name,patient_first_name,patient_dob,patient_city,patient_state)"
+              - "Bundle.identifier.value.empty().not()"
+              - "Bundle.entry.resource.ofType(DiagnosticReport).result.resolve().where(method.empty() or value.coding.code.empty()).count() = 0"
+              - "Bundle.entry.resource.ofType(Specimen).where(type.empty()).count() = 0"
+              - "Bundle.entry.resource.ofType(Patient).where(name.family.empty() or name.given.empty() or birthDate.empty()).count() = 0"
+              - "Bundle.entry.resource.ofType(Patient).where(address.city.empty()).count() = 0"
+              - "Bundle.entry.resource.ofType(Patient).where(address.state.empty()).count() = 0"
+              # "isValidCLIA(testing_lab_clia,reporting_facility_clia)"
+              - "Bundle.entry.resource.ofType(ServiceRequest).performer.resolve().identifier.where(type.coding.code = 'CLIA').value.getIdType() = 'CLIA'"
+              - "Bundle.entry.resource.ofType(ServiceRequest).requester.resolve().organization.resolve().identifier.where(type.coding.code = 'CLIA').value.getIdType() = 'CLIA'"
+      routingFilter:
+      #        # "matches(abnormal_flag, A)"
+              - "Bundle.entry.resource.ofType(Observation).interpretation.coding.code = 'A'"
+      #        # "matches(test_result_status, F, S, P, C, X, I)"
+              - "Bundle.entry.resource.ofType(Observation).status = 'final' or Bundle.entry.resource.ofType(Observation).status = 'preliminary' or Bundle.entry.resource.ofType(Observation).status = 'final' or Bundle.entry.resource.ofType(Observation).status = 'amended' or Bundle.entry.resource.ofType(Observation).status = 'corrected'"
```

Comment:
Use the following:
        # - "matches(test_result_status, F, S, P, C, X, I)"
        # - "matches(test_result_status, F=final, S=partial, P=preliminary, C=corrected, X=canceled, I=incomplete)"
        # see: https://hl7-definition.caristix.com/v2/HL7v2.5.1/Tables/0085, https://www.hl7.org/fhir/valueset-diagnostic-report-status.html )"
        - "Bundle.entry.resource.ofType(DiagnosticReport).where(status in 'final'|'partial'|'preliminary'|'corrected'|'canceled'|'incomplete').exists()"

Reply:
I did see the two and that is not how I interpreted the mappings. I was waiting for @victor-chaparro to verify what the mapping from one to the other should be, but I do appreciate the shorter way of handling it. They are not 1:1 in the FHIR mapping https://build.fhir.org/valueset-observation-status.html